### ISSUE TITLE
opendmarc-report issue using GROUP BY

### ISSUE THREAD

**Issue by tuxtuxdm** [trusteddomainproject/OpenDMARC#249]
Hi,

it seems for mysql backend there is a problem with this query:

```
$dbi_s = $dbi_h->prepare("SELECT repuri, adkim, aspf, requests.policy, spolicy, pct, UNIX_TIMESTAMP(lastsent), domains.name FROM requests JOIN messages ON messages.from_domain=requests.domain LEFT JOIN domains ON messages.policy_
domain = domains.id WHERE domain = ? GROUP BY policy_domain");
```

See:

https://dev.mysql.com/doc/refman/8.0/en/group-by-handling.html

I've fixed it with:

```
445c445
<       $dbi_s = $dbi_h->prepare("SELECT repuri, adkim, aspf, requests.policy, spolicy, pct, UNIX_TIMESTAMP(lastsent), domains.name FROM requests JOIN messages ON messages.from_domain=requests.domain LEFT JOIN domains ON messages.policy_domain = domains.id WHERE domain = ? GROUP BY policy_domain,repuri,adkim,aspf,requests.policy,spolicy,pct,domains.name,lastsent");
---
>       $dbi_s = $dbi_h->prepare("SELECT repuri, adkim, aspf, requests.policy, spolicy, pct, UNIX_TIMESTAMP(lastsent), domains.name FROM requests JOIN messages ON messages.from_domain=requests.domain LEFT JOIN domains ON messages.policy_domain = domains.id WHERE domain = ? GROUP BY policy_domain");
```

Best regards,

Dani.

**Comment by thegushi** [comment]
Good catch, but this query no longer exists in either `master` or `develop` — it was simplified in a prior refactor and the complex JOIN/GROUP BY was replaced with a straightforward `SELECT ... FROM requests WHERE domain = ?`. This should be resolved if you upgrade to the current release. Closing, but feel free to reopen if you're still seeing this on a current version.

**Comment by tuxtuxdm** [comment]
Thanks for the hint.
Yeah, it can be my Debian version... not being up to date with the source code...
There is no update available for my current Debian 12.

**Comment by thegushi** [comment]
> Thanks for the hint. Yeah, it can be my Debian version... not being up to date with the source code... There is no update available for my current Debian 12.

Okay, so I'll close this one out.  I'll note though that these are perl scripts. You can totally grab the openDMARC source and pull the perl scripts out directly and put them in a different dir, without needing to mess with your OS packages.  You don't even need a compiler.

I've honestly thought about breaking the reporting functions out to a different module, for organizations that want to only do that on functions where they're not running a the milters.
